src/main/java/testapp/adapter/http/UserService.java

```diff
@@ -10,39 +10,58 @@
 import com.google.sitebricks.http.Get;
 import com.google.sitebricks.http.Post;
 import testapp.core.User;
-import testapp.core.UserAlreadyExistsException;
+import testapp.core.UserException;
 import testapp.core.UserRepository;
 
 import java.net.HttpURLConnection;
 import java.util.List;
 
 @At("/users")
 public class UserService {
+    private UserRepository userRepository;
 
     @Inject
-    private UserRepository repository;
+    public UserService(UserRepository repository) {
+        this.userRepository = repository;
+    }
 
     @Get
     public Reply<?> getUsers() {
-        List<User> userList = repository.findUsers();
+        List<User> userList = userRepository.findUsers();
         return Reply.with(userList).as(Json.class);
     }
 
+    @Get
+    @At("/:userId")
+    public Reply<?> findUserById(@Named("userId") int userId) {
+        User user;
+        try {
+            user = userRepository.findById(userId);
+        } catch (UserException e) {
+            return Reply.saying().badRequest();
+        }
+        return Reply.with(user).as(Json.class);
```

Comment:
I would suggest using a Dto for adapter between the app and the client app, and have a separate domain object.
*Dto(Data Transfer Object) containing only fields of the domain object without the logic, used only for transportation of the data.